File: app/src/main/java/com/example/opencapprovidertest/VehicleInformationProvider.java
package com.example.opencapprovidertest;

import android.util.Log;

import org.opencabstandard.provider.VehicleInformationContract;

public class VehicleInformationProvider extends AbstractVehicleInformationProvider {
    private static final String LOG_TAG = "VehicleInformationProvider";

    @Override
    public VehicleInformationContract.VehicleInformation getVehicleInformation(String version) {
        Log.d(LOG_TAG, "getVehicleInformation()");

        VehicleInformationContract.VehicleInformation vehicleInformation = new VehicleInformationContract.VehicleInformation();
        if (CurrentVin.getInstance().getVin() != null) {
            vehicleInformation.setVin(CurrentVin.getInstance().getVin());
        }
        vehicleInformation.setVehicleId("Great Vehicle ID 1");
        vehicleInformation.setInGear(true);

        printVehicleInformation(vehicleInformation);
        return vehicleInformation;
    }

    private void printVehicleInformation(VehicleInformationContract.VehicleInformation vehicleInformation) {
        String vinIsNullSuffix = "false";
        if (vehicleInformation.getVin() == null) {
            vinIsNullSuffix = "true";
        }

        Log.d(LOG_TAG, "VIN IS NULL" + vinIsNullSuffix);
    }
}

File: app/src/main/java/com/example/opencapprovidertest/opencab/VehicleInformationProvider.kt
package com.example.opencapprovidertest.opencab

import android.util.Log
import com.example.opencapprovidertest.AbstractVehicleInformationProvider
import com.example.opencapprovidertest.VehicleInformationContract.VehicleInformation


class VehicleInformationProvider : AbstractVehicleInformationProvider() {
    override fun getVehicleInformation(version: String): VehicleInformation {
        Log.d(LOG_TAG, "getVehicleInformation()")

        val vehicleInformation = VehicleInformation()
        vehicleInformation.vehicleId = "Great Vehicle ID 1"
        vehicleInformation.vin = "QWERRTYUIOP12345"
        vehicleInformation.isInGear = true
        return vehicleInformation
    }

    companion object {
        private val LOG_TAG: String = "OpenCabVehicleInfoProvider"
    }
}